engine/src/main/java/org/kigalisim/engine/state/OverridingConverterStateGetter.java

```diff
@@ -1,136 +1,265 @@
 /**
- * Interface for overriding converter state getter.
+ * State getter that allows overriding values from an inner state getter.
  *
- * <p>This interface will be used for the future transition of converter state
- * management. It defines the contract for state getters that can override
- * values from an inner state getter.</p>
+ * <p>This class wraps an inner StateGetter and allows individual values to be
+ * overridden while delegating to the inner getter for non-overridden values.</p>
  *
  * @license BSD-3-Clause
  */
 
 package org.kigalisim.engine.state;
 
+import java.util.HashMap;
+import java.util.Map;
 import org.kigalisim.engine.number.EngineNumber;
 import org.kigalisim.engine.number.UnitConverter;
 
 /**
- * Interface for state getter that allows overriding values.
+ * State getter that allows overriding values from an inner state getter.
  *
- * <p>This interface defines methods for retrieving and overriding various
- * state values needed for unit conversions and calculations in the engine.</p>
+ * <p>This class wraps an inner StateGetter and allows individual values to be
+ * overridden while delegating to the inner getter for non-overridden values.</p>
  */
-public interface OverridingConverterStateGetter {
+public class OverridingConverterStateGetter implements StateGetter {
+
+  private final StateGetter innerGetter;
+  private EngineNumber energyIntensity;
```

Comment:
Please use Optional instead of null for all of these values across the file. Instead of check if equal to null, check if empty.

```
  private EngineNumber energyIntensity;
  private EngineNumber amortizedUnitVolume;
  private EngineNumber population;
  private EngineNumber yearsElapsed;
  private EngineNumber totalConsumption;
  private EngineNumber energyConsumption;
  private EngineNumber volume;
  private EngineNumber amortizedUnitConsumption;
  private EngineNumber populationChange;
  private EngineNumber substanceConsumption;
```